xdsl/ir.py

```diff
@@ -475,6 +489,13 @@ class Operation(IRNode):
     parent: Block | None = field(default=None, repr=False)
     """The block containing this operation."""
 
+    traits: frozenset[OpTrait] = field(init=False)
```

Comment:
Should this not be `ClassVar`?

Reply:
Hmm, that's a good question. For unparametrized traits, it seems that they should definitely be on the class, not sure if there are any cases where we'd want them to be per instance.

Reply:
Thank god for pointing me to it! I tried to find this for so long...